### PR TITLE
[storage] Get metadata file list before syncing and downloading

### PR DESCRIPTION
**Pull request by areshand**
### Description
Get a snapshot of remote metadata files before syncing the metadata file.

There is a corner case that syncing metadata file can run a long time and the snapshot of remote files can contain new files that are not covered by compaction

### Test Plan
Test on GCP with compaction job and backup verify

### CODE REVIEW

File: storage/backup/backup-cli/src/coordinators/backup.rs [hunk 1]
```diff
@@ -358,6 +358,9 @@ impl BackupCompactor {
 
     pub async fn run(&self) -> Result<()> {
         info!("Backup compaction started");
+        // Get a snapshot of remote metadata files before compaction
+        let files = self.storage.list_metadata_files().await?;
+
         // sync the metadata from backup storage
         let mut metaview = metadata::cache::sync_and_load(
```
Comment (msmouse) [inline storage/backup/backup-cli/src/coordinators/backup.rs:365]: I'd rather return the file list from here -- no chance of inconsistency then.

Comment (areshand) [reply storage/backup/backup-cli/src/coordinators/backup.rs]: updated to return the file list from metaview

**Review by grao1991 — APPROVED** [review]
